dio/lib/src/dio.dart

```diff
@@ -935,12 +940,12 @@ abstract class DioMixin implements Dio {
         statusMessage: responseBody.statusMessage,
         extra: responseBody.extra,
       );
-      var statusOk = options.validateStatus(responseBody.statusCode);
-      if (statusOk || options.receiveDataWhenStatusError) {
+      var statusOk = options.validateStatus!(responseBody.statusCode);
+      if (statusOk || options.receiveDataWhenStatusError != null) {
```

Comment:
The check for the actual value of `options.receiveDataWhenStatusError` is missing
```suggestion
      if (statusOk || options.receiveDataWhenStatusError == true) {
```